### PR TITLE
Hide suggestion popup when changing active window

### LINKED ISSUE #1606: Opening new tab (using shortcut) does not close search suggestion popup
Steps to reproduce:
1. Write something in address bar or search bar; suggestion popup appears
2. Press 'Open tab' shortcut (default CTRL+T)
 
Result:
Suggestion popup stays open, you can continue writing into it.

Expected:
Suggestion popup should be hided.

Video demo:
https://www.youtube.com/watch?v=ofJNedKG00E&feature=youtu.be

### PR DESCRIPTION
**Pull request by fnkkio**
Fixes #1606

### CODE REVIEW

File: src/modules/widgets/search/SearchWidget.cpp [hunk 1]
```diff
@@ -753,6 +753,7 @@ void SearchWidget::setWindow(Window *window)
 		disconnect(this, &SearchWidget::requestedSearch, m_window.data(), &Window::requestedSearch);
 		disconnect(m_window.data(), &Window::loadingStateChanged, this, &SearchWidget::handleLoadingStateChanged);
 		disconnect(m_window.data(), &Window::optionChanged, this, &SearchWidget::handleWindowOptionChanged);
+		disconnect(window, &Window::hidden, this, &SearchWidget::hidePopup);
```
Comment (Emdek) [inline src/modules/widgets/search/SearchWidget.cpp]: It should be `m_window.data()`.

**Comment by fnkkio** [comment]
@Emdek Ops.. fixed.

**Comment by Emdek** [comment]
@fnkkio, I'm not sure if this is the right approach, it might not work in case of MDI, where several could be visible at the same time.
Maybe `MainWindow::currentWindowChanged()` would be more suitable?

BTW, better place to add such signal would be `Window::hideEvent()` anyway.

**Comment by fnkkio** [comment]
@Emdek You are right, this would not work for MDI interface. Updated according to your suggestions.

**Pull request merged**